monkestation/code/game/objects/items/implants/implant_hardspear.dm

```diff
@@ -0,0 +1,78 @@
+/obj/item/implant/hardspear
+	name = "hardlight spear implant"
+	icon = 'monkestation/icons/obj/implants.dmi'
+	icon_state = "lightspear" //Shows up as the action button icon
+	implant_color = "b"
+	uses = -1
+	COOLDOWN_DECLARE(hardlight_implant_cooldown)
+
+/obj/item/implant/hardspear/activate()
+	if(!COOLDOWN_FINISHED(src, hardlight_implant_cooldown)) //Thanks implant_abductor.dm for the help <3
+		to_chat(imp_in, "<span class='warning'>You must wait [COOLDOWN_TIMELEFT(src, hardlight_implant_cooldown)*0.1] seconds to use [src] again!</span>")
+		return
+
+	var/obj/item/spear/awesomespear
```

Comment:
You can use the typepath of the hard light spear directly.
new var/obj/item/spear/hardlightspear/summoned_spear(loc) for example.

Also, change the variable name to something more relevant for readability